sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -120,17 +120,23 @@ public virtual TokenCredential CreateWorkloadIdentityCredential()
 
         public virtual TokenCredential CreateManagedIdentityCredential()
         {
-            return new ManagedIdentityCredential(new ManagedIdentityClient(
-                new ManagedIdentityClientOptions
-                {
-                    ResourceIdentifier = Options.ManagedIdentityResourceId,
-                    ClientId = Options.ManagedIdentityClientId,
-                    Pipeline = Pipeline,
-                    Options = Options,
-                    InitialImdsConnectionTimeout = TimeSpan.FromSeconds(1),
-                    ExcludeTokenExchangeManagedIdentitySource = Options.ExcludeWorkloadIdentityCredential
-                })
-            );
+            var options = Options.Clone<DefaultAzureCredentialOptions>();
+            // Set the custom retry policy
+            options.Retry.MaxRetries = 4;
+            options.RetryPolicy ??= new DefaultAzureCredentialImdsRetryPolicy(options.Retry);
```

Comment:
The changelog said you didn't retry - or is this used differently?

Reply:
We only don't retry on the initial probe request (and only when done from DAC as opposed to `ManagedIdentityCredential` directly.

Reply:
I saw that later in the PR. Thanks for confirming.